### PR TITLE
kubeadm: Remove .AuthorizationModes in the v1alpha2 API

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:
Now that we have https://github.com/kubernetes/kubernetes/pull/63879, we don't actually need to have `:AuthorizationModes` in our API anymore. This PR removes support for `.AuthorizationModes` in the v1alpha2 API, but keeps an upgrade path available (automatic conversion) from the v1alpha1 version.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of kubernetes/community#2131

**Special notes for your reviewer**:
Depends on:
 - [x] https://github.com/kubernetes/kubernetes/pull/63879
 - [x] https://github.com/kubernetes/kubernetes/pull/63917

**Release note**:

```release-note
[action required] kubeadm: Support for `.AuthorizationModes` in the kubeadm v1alpha2 API has been removed. Instead, you can use the `.APIServerExtraArgs` and `.APIServerExtraVolumes` fields to achieve the same effect. Files using the v1alpha1 API and setting this field will be automatically upgraded to this v1alpha2 API and the information will be preserved.
```
@kubernetes/sig-cluster-lifecycle-pr-reviews @liztio

### CODE REVIEW

**Comment by luxas** [comment]
rebased upon #63879 and #63917 :+1:

**Comment by neolit123** [comment]
this page has a `v1alpha1` example:
https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-init/#config-file

should it be updated to `v1alpha2` for 1.11?

**Review by timothysc — APPROVED** [review]
/lgtm 

@liztio - we're going to need to doc the args feedback loop, b/c I can guarantee there will be feedback.

File: cmd/kubeadm/app/util/config/testdata/defaulting/master/defaulted_v1alpha1.yaml [hunk 1]
```diff
@@ -1,78 +0,0 @@
-api:
```
Comment (liztio) [inline cmd/kubeadm/app/util/config/testdata/defaulting/master/defaulted_v1alpha1.yaml]: this change seems unrelated?

Comment (luxas) [reply cmd/kubeadm/app/util/config/testdata/defaulting/master/defaulted_v1alpha1.yaml]: It's https://github.com/kubernetes/kubernetes/pull/64068/commits/4f0020d1b4caec992460859ceb792f1b785a85fd, I had to remove it as I realized kubeadm v1.11 won't be able to marshal the v1alpha1 API, only the newer v1alpha2 (this due to that conversions only work one way: from v1alpha1 to v1alpha2(==internal), not the other way round).
Hence I had to remove the e2e unit tests that printed v1alpha1 versions as they won't work.

**Comment by luxas** [comment]
> we're going to need to doc the args feedback loop, b/c I can guarantee there will be feedback.

Yup. I also filed https://github.com/kubernetes/kubeadm/issues/842 so we can take advantage of the godoc comments as much as possible initially, and then write custom docs on top, instead of just the limited amount of manually created content we have right now. As we grow more confident in our API we can also start documenting it more clearly and visibly.

/retest

**Comment by luxas** [comment]
> should it be updated to v1alpha2 for 1.11?

@neolit123 yes, but we'd like to have automatically-generated content as a base, and add custom API docs on top.

**Comment by neolit123** [comment]
> yes, but we'd like to have automatically-generated content as a base, and add custom API docs on top.

i understand,
that config example is not generated, so it has to be edited manually each time the API changes.
i'm not sure what tooling will generate such an example config that also dumps the types:
https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-init/#config-file
